File: PythonTests/Cucumber-Behave/steps/google_search_steps.py
from behave import given, when, then
from selenium import webdriver
from selenium.webdriver.common.by import By
import time

@given('I open Google homepage')
def step_open_google(context):
    context.driver = webdriver.Chrome()
    context.driver.get("https://www.google.com")

@when('I search for "{query}"')
def step_search(context, query):
    context.driver.find_element(By.XPATH, "//*[@id=\"L2AGLb\"]/div").click()
    search_box = context.driver.find_element(By.NAME, "q")
    search_box.send_keys(query)
    search_box.submit()

@then('the page title should contain "{query}"')
def step_check_title(context, query):
    time.sleep(2)  # simple wait
    title = context.driver.title
    assert query in title
    context.driver.quit()

File: CSharpTests/specflow/Steps/GoogleSearchSteps.cs
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using TechTalk.SpecFlow;
using System.Threading;


    [Binding]
    public class GoogleSearchSteps
    {
        private IWebDriver driver;

        [Given(@"I open Google homepage")]
        public void GivenIOpenGoogleHomepage()
        {
            driver = new ChromeDriver();
            driver.Navigate().GoToUrl("https://www.google.com");
        }

        [When(@"I search for ""(.*)""")]
        public void WhenISearchFor(string query)
        {
            driver.FindElement(By.xpath("//*[@id=\"L2AGLb\"]/div")).Click();

            var searchBox = driver.FindElement(By.Name("q"));
            searchBox.SendKeys(query);
            searchBox.Submit();
        }

        [Then(@"the page title should contain ""(.*)""")]
        public void ThenThePageTitleShouldContain(string query)
        {
            Thread.Sleep(2000);
            Assert.IsTrue(driver.Title.Contains(query));
            driver.Quit();
        }
    }

